### ISSUE TITLE
Remove portal_properties Tool.

### ISSUE THREAD

**Issue by tisto** [plone/Products.CMFPlone#125]
Once all settings have been moved to plone.app.registry, we should be able to remove portal_properties.

**Comment by esteele** [comment]
All of the properties currently set by Plone core have been migrated to the configuration registry. I think we can mark PropertiesTool as deprecated in 5.0 and remove in 5.1.

**Comment by vangheem** [comment]
Moved from plip to bug. We need to mark it as deprecated and fully remove this in 6 IMO.

**Comment by pbauer** [comment]
How do we properly deprecate portal_properties? ONe option is to simplyt mention it in the release notes. Another would be to throw deprecation warnings when accessing properties or creating propertysheets. (i.e. decorating addPropertySheet, _setPropValue and getProperty)

**Comment by mauritsvanrees** [comment]
When running some tests, I saw this warning again:

```
DeprecationWarning:
The portal portal_properties tool will be removed in Plone 6.1.
Use the portal_registry instead.
Check https://github.com/plone/Products.CMFPlone/issues/125 for more details.
```

I have added the current issue to the Plone 6.1 project board.

Grepping for `portal_properties` only in Python code already pulls up a lot of lines.  Some will be for backward compatibility and can simply be removed.  I hope this is true for all.

```
$ ack --python portal_properties
Products/CMFPlone/tests/testPortalCreation.py
45:        self.properties = self.portal.portal_properties
951:        self.properties = self.portal.portal_properties

Products/CMFPlone/tests/testNavigationView.py
126:        ntp = self.portal.portal_properties.navtree_properties
137:        ntp = self.portal.portal_properties.navtree_properties

Products/CMFPlone/tests/test_zmi.py
124:    def test_portal_properties(self):
125:        url = "portal_properties/manage_main"

Products/CMFPlone/tests/testWebDAV.py
23:        ptool = getToolByName(self.portal, "portal_properties")

Products/CMFPlone/PropertiesTool.py
28:    id = "portal_properties"
84:    #   'portal_properties' interface methods
107:        "The portal portal_properties tool will be removed in Plone 6.1. "
135:        "The portal portal_properties tool will be removed in Plone 6.1. "

Products/CMFPlone/setuphandlers.py
78:        "portal_properties": "General settings registry",

Products/CMFPlone/browser/author.py
200:        self.portal_properties = getUtility(IPropertiesTool)

Products/CMFPlone/exportimport/tests/testPropertiesTool.py
18:<object name="portal_properties" meta_type="Plone Properties Tool">

Products/CMFPlone/exportimport/propertiestool.py
25:    ptool = getToolByName(site, "portal_properties")
44:    ptool = getToolByName(site, "portal_properties", None)

Products/CMFCore/interfaces/_tools.py
1089:                   o Must be set to 'portal_properties'.

plone/app/portlets/portlets/navigation.py
409:        portal_properties = getToolByName(context, "portal_properties")
410:        navtree_properties = getattr(portal_properties, "navtree_properties")

plone/app/vocabularies/types.py
44:    portal_properties = getToolByName(context, "portal_properties", None)
45:    if portal_properties is not None:
47:    site_properties = getattr(portal_properties, "site_properties", None)
119:      >>> tool = DummyTool('portal_properties')
127:      >>> context.portal_properties = tool

plone/app/upgrade/tests/base.py
64:        # Removes a site property from portal_properties
65:        tool = getToolByName(self.portal, "portal_properties")
71:        # adds a site property to portal_properties
72:        tool = getToolByName(self.portal, "portal_properties")
78:        # Removes a navtree property from portal_properties
79:        tool = getToolByName(self.portal, "portal_properties")
85:        # adds a navtree property to portal_properties
86:        tool = getToolByName(self.portal, "portal_properties")

plone/app/layout/globals/interfaces.py
79:        """The portal_properties tool"""

plone/app/layout/globals/tools.py
26:        return getToolByName(self.context, "portal_properties")

plone/app/layout/globals/tests/test_tools.py
34:            self.view.properties(), getToolByName(self.folder, "portal_properties")

plone/app/layout/sitemap/tests/test_sitemap.py
48:            self.portal, "portal_properties"

plone/app/textfield/utils.py
26:    """Get a set of allowed MIME types according to the portal_properties
42:        portal_properties = getToolByName(site, "portal_properties", None)
43:        if portal_properties is None:
46:        site_properties = portal_properties.get("site_properties", None)

plone/app/textfield/tests.py
299:        self.portal["portal_properties"]["site_properties"]._setPropValue(
333:        self.portal["portal_properties"]["site_properties"]._setPropValue(

plone/app/querystring/tests/testQueryParser.py
107:        self.portal_properties = MockSiteProperties()

plone/app/users/browser/registersettingspanel.py
80:        pprop = getToolByName(self.context, "portal_properties")

plone/app/contenttypes/setuphandlers.py
135:    portal_properties = getToolByName(portal, "portal_properties")
136:    site_properties = portal_properties.site_properties

plone/restapi/services/contextnavigation/get.py
628:        portal_properties = getToolByName(context, "portal_properties")
629:        navtree_properties = getattr(portal_properties, "navtree_properties")

plone/api/tests/test_user.py
51:            portal.portal_properties.site_properties.use_email_as_login = value

plone/api/tests/test_portal.py
124:            # before Plone 4.3, date formats were stored in portal_properties
125:            properties = portal.get_tool("portal_properties")

plone/dexterity/content.py
70:    "portal_properties",

plone/base/navigationroot.py
18:    portal_properties, navigation root path is computed from
35:        # fetch from portal_properties

plone/base/interfaces/properties.py
9:    id = Attribute("id", 'Must be set to "portal_properties"')
```

**Comment by jensens** [comment]
it worth the effort and would clean our code.

**Comment by mauritsvanrees** [comment]
All done, and released in Plone 6.1.0a4 two weeks ago.
Closing issue 125. :-)
